Friendly-plans/app/src/main/java/pg/autyzm/friendly_plans/view/components/SoundComponent.java

```diff
@@ -0,0 +1,109 @@
+package pg.autyzm.friendly_plans.view.components;
+
+import android.content.Context;
+import android.media.AudioManager;
+import android.media.MediaPlayer;
+import android.view.View;
+import android.view.animation.Animation;
+import android.view.animation.AnimationUtils;
+import android.widget.ImageView;
+import database.entities.Asset;
+import database.repository.AssetRepository;
+import java.io.IOException;
+import javax.inject.Inject;
+import pg.autyzm.friendly_plans.AppComponent;
+import pg.autyzm.friendly_plans.R;
+import pg.autyzm.friendly_plans.asset.AssetsHelper;
+import pg.autyzm.friendly_plans.notifications.ToastUserNotifier;
+
+public final class SoundComponent {
+
+    @Inject
+    MediaPlayer mediaPlayer;
+    @Inject
+    AssetRepository assetRepository;
+
+    private AssetsHelper assetsHelper;
+    private Long soundId;
+    private ImageView playSoundIcon;
+    private Context context;
+
+    public static SoundComponent getSoundComponent(Long soundId, ImageView playSoundIcon,
+            Context context, AppComponent appComponent) {
+        final SoundComponent soundComponent = new SoundComponent(soundId, playSoundIcon, context);
+        appComponent.inject(soundComponent);
```

Comment:
This is odd..

Reply:
Yes, but unfortunately it's problematic to solve it in a different way... SoundComponent creation process consists:
- create new instance of SoundComponent with dependencies from Fragment (soundId, icon)
- inject Dagger managed dependencies into a brand new instance
- configure media player (set audioStreamType and onCompletionListener)

Those must be done in this order because injecting requires an instance to which we are injecting and configuration of media player must be done when it exists, so after it has been injected into the SoundComponent...

I know it's a bit wired but from the SoundComponent user perspective (e.g. Fragment) whole creation process is not so important - it just always gives the right instance :) Maybe in the future, we will find some way to fix it :)